src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -224,19 +237,21 @@ public PolicyEntity getPolicy(@PathParam("domainid") Long domainid, @PathParam("
 
             @Override
             public PolicyEntity call() throws Exception {
-                CriteriaBuilder cb = em.getCriteriaBuilder();
-                CriteriaQuery<PolicyEntity> q = cb.createQuery(PolicyEntity.class);
-                Root<PolicyEntity> r = q.from(PolicyEntity.class);
-                q.select(r)
-                        .where(cb.and(cb.equal(r.get("domain").get("Id"), domainid), cb.equal(r.get("id"), policyid)));
-                List<PolicyEntity> result = em.createQuery(q).getResultList();
-                if (result.isEmpty() == false) {
-                    PolicyEntity policy = new PolicyEntity();
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<PolicyEntity> query = criteriaBuilder.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = query.from(PolicyEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("domain").get("Id"), domainid),
+                        criteriaBuilder.equal(r.get("id"), policyid)));
+                List<PolicyEntity> result = em.createQuery(query).getResultList();
```

Comment:
just return the result

Reply:
ok result.get(0)

Reply:
after testing I see that returning result.get(0) results in exception, so reverted the changes and did not find time to debug. Also returning entity while persisting to DB results in an exception due to DB persists goes in background and then the data is returned, meanwhile result is already published.

Reply:
will root cause the first issue and add comment for the second in the subsequent user stories.